File: system-test/dotnet/SystemTests/Legacy/Mod03/E2eTests/Base/BaseE2eTest.cs
using Dsl.Port;
using Dsl.Core.Shared;
using ConfigEnvironment = SystemTests.TestInfrastructure.Configuration.Environment;
using SystemTests.TestInfrastructure.Configuration;
using SystemTests.Legacy.Mod03.Base;

namespace SystemTests.Legacy.Mod03.E2eTests.Base;

public abstract class BaseE2eTest : BaseRawTest
{
    protected override ExternalSystemMode? GetFixedExternalSystemMode()
    {
        return ExternalSystemMode.Real;
    }

    public override async Task InitializeAsync()
    {
        await SetShopRawAsync();
        SetUpExternalHttpClients();
    }

    protected abstract Task SetShopRawAsync();

    protected static string CreateUniqueSku(string baseSku)
    {
        var suffix = Guid.NewGuid().ToString("N")[..8];
        return $"{baseSku}-{suffix}";
    }
}

File: system-test/java/src/test/java/com/optivem/shop/systemtest/legacy/mod07/e2e/base/BaseE2eTest.java
package com.optivem.shop.systemtest.legacy.mod07.e2e.base;

import com.optivem.shop.systemtest.legacy.mod07.base.BaseUseCaseDslTest;
import com.optivem.shop.systemtest.configuration.ExternalSystemMode;

public abstract class BaseE2eTest extends BaseUseCaseDslTest {
    @Override
    protected ExternalSystemMode getFixedExternalSystemMode() {
        return ExternalSystemMode.REAL;
    }
}




